tests/by-util/test_du.rs

```diff
@@ -1220,3 +1220,65 @@ fn test_human_size() {
         .succeeds()
         .stdout_contains(format!("1.0K	{dir}"));
 }
+
+#[cfg(not(target_os = "android"))]
+#[test]
+fn test_du_deduplicated_input_args() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("d");
+    at.mkdir("d/d");
+    at.touch("d/f");
+    at.hard_link("d/f", "d/h");
+
+    let result = ts
+        .ucmd()
+        .arg("--inodes")
+        .arg("d")
+        .arg("d")
+        .arg("d")
+        .succeeds();
+    result.no_stderr();
+
+    let mut result_seq: Vec<String> = result
+        .stdout_str()
+        .split('\n')
```

Comment:
I think `lines()` is more expressive:
```suggestion
        .lines()
```

Reply:
@samueltardieu one more thing that clippy could detect? :)

Reply:
@sylvestre This is not strictly equivalent, as "foo\r\nbar" will be split as ["foo\r", "bar"] while lines() would return ["foo", "bar"] (it splits on both "\r\n" and "\n", but not "\r").